### PR TITLE
Use the Beaker release API to download the CLI and remove the dependency on a GH token

### PR DESCRIPTION
**Pull request by mewil**
Fixes #

Changes proposed in this pull request:

- Updates the action to use the beaker release API which removes the need to a GH token. This seems to be a breaking change, so I've bumped the version to `2.0.0`.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/allenai/setup-beaker/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/allenai/setup-beaker/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [x] If this PR fixes a bug, I've added a test that will fail without my fix.
- [x] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [v2.0.0](https://github.com/allenai/setup-beaker/releases/tag/v2.0.0) - 2022-08-15
```
Comment (epwalsh) [inline CHANGELOG.md:10]: This line actually gets added automatically during Step 2 of the [release process](https://github.com/allenai/setup-beaker/blob/main/RELEASE_PROCESS.md).

Comment (epwalsh) [reply CHANGELOG.md:10]:
```suggestion
```

**Review by epwalsh — COMMENTED** [review]
Looks like you'll need to remove [this line](https://github.com/allenai/setup-beaker/blob/ef55361568ccf496558ced70675c670f50c0377a/.github/workflows/main.yml#L52) to make CI pass

**Comment by mewil** [comment]
> Looks like you'll need to remove [this line](https://github.com/allenai/setup-beaker/blob/ef55361568ccf496558ced70675c670f50c0377a/.github/workflows/main.yml#L52) to make CI pass

Will do! This change also exposes that the release API is broken due to some GH SAML auth changes, so this can't be merged yet.

File: CHANGELOG.md (hunk 1 above)
Comment (mewil) [reply CHANGELOG.md]: Oops, thanks!

**Review by epwalsh — APPROVED** [review]
LGTM! Feel free to merge once CI passes

**Pull request merged**